### ISSUE TITLE
Chrome issue - connects with null protocol header

### ISSUE THREAD

**Issue by exabrial** [porsager/pws#8]
Is there a way to use this without node?

**Comment by porsager** [comment]
Do you mean in the browser?

**Comment by porsager** [comment]
Let me know or reopen if this wasn't the answer you looked for ;) I'll close this for now.

**Comment by exabrial** [comment]
Apologies.

So lets say you have an app that doesn't use node or a package manager. Is using PWS as simple as including the index.js you linked above?

**Comment by porsager** [comment]
Ah yes, that's correct ;)

So you can do
```
<script src="pws.js"></script>
<script>
  const socket = new PersistentWebSocket(url)
  // More code here
</script>
```
[]

**Comment by exabrial** [comment]
Awesome thanks. 

We're doing the following:
```
var	websocket;
function connectSocket() {
	var wsUrl = 'ws://localhost:8080/something';
	websocket = new PersistentWebSocket(wsUrl, null, {
		reachabilityTestUrl: "/haproxy.jsp" 
	});
	window.onbeforeunload = function() {
	    websocket.onclose = function() {};
	    websocket.close();
	};
	websocket.onmessage = (event) => {
		if (event.data === 'update') {
			updateTexts();
		} else if (event.data === 'ping?') {
			websocket.send('pong!');
		} else if (event.data === 'optout') {
			forceOptout = true;
			receivedOptInOutStatus = true;
			enableDisableOptOutStatusIndicator();
		} else if (event.data === 'optin') {
			forceOptout = false;
			receivedOptInOutStatus = true;
			enableDisableOptOutStatusIndicator();
		}
	};
	websocket.open();
}
```

And getting this error in the console:
`WebSocket connection to 'ws://localhost:8080/something' failed: Error during WebSocket handshake: Sent non-empty 'Sec-WebSocket-Protocol' header but no response was received`

**Comment by porsager** [comment]
Yes, browsers set the header by default unless you specify one, so your server should handle that. This is not really related to pws, but websockets In general.

**Comment by exabrial** [comment]
interesting, why does PWS fail? If we use plain websocket it doesn't do this. Is there a way to disable that behavior?

**Comment by porsager** [comment]
Does it work if you don't supply the extra arguments to pws?

**Comment by porsager** [comment]
Just give it the URL and nothing else

**Comment by exabrial** [comment]
I did, same problem! But changing
```
websocket = new PersistentWebSocket(wsUrl);
```
to 
```
websocket = new WebSocket(wsUrl);
```

Works! Could it be related to TLS somehow? I notice all of the examples are using it, whereas locally we don't have it running

**Comment by exabrial** [comment]
Actually check this out, this is with PWS:

![Screen Shot 2019-08-30 at 2 20 40 PM](https://user-images.githubusercontent.com/1392297/64046500-99761700-cb31-11e9-9a5e-248fbcd28b57.png)

**Comment by exabrial** [comment]
this is with plain websocket:
![Untitled](https://user-images.githubusercontent.com/1392297/64046521-a430ac00-cb31-11e9-847c-6683f626716b.png)

**Comment by porsager** [comment]
Ah right, you might be on to something! Let me run a quick test

**Comment by porsager** [comment]
Ahha! Chrome changed the behavior of new WebSocket! :-(

**Comment by porsager** [comment]
I'll make a fix immidiately

**Comment by porsager** [comment]
Thanks a lot for the report :) It's fixed in v4.1.0, just released.

**Comment by exabrial** [comment]
Hah you bet! thank YOU for helping me out
